program-runtime/src/loaded_programs.rs

```diff
@@ -416,6 +423,20 @@ impl LoadedProgram {
             && slot >= self.deployment_slot
             && slot < self.effective_slot
     }
+
+    pub fn update_access_slot(&self, slot: Slot) {
+        let _ = self.latest_access_slot.fetch_update(
+            Ordering::Relaxed,
+            Ordering::Relaxed,
+            |last_access| (last_access < slot).then_some(slot),
+        );
+    }
+
+    fn decayed_usage_counter(&self, now: Slot) -> u64 {
```

Comment:
This should be used for the sorting at the beginning of the recompilation phase as well.